src/sas/sasgui/perspectives/fitting/fitting.py

```diff
@@ -1875,6 +1892,9 @@ def _complete2D(self, image, data, model, page_id, elapsed, index, qmin,
                 self._plot_residuals(page_id=page_id, data=data, fid=fid,
                                       index=index, weight=weight)
         msg = "Computation  completed!"
+        if number_finite != len(image):
```

Comment:
if `y` is not 1 dimensional (perhaps not out of bounds of possibility) then this statement will be True.
Use `if number_finite != y.size`